Documentation/PropertyWrapper/Parameter.md

```diff
@@ -280,110 +265,115 @@ Exposing the `Component` on a web socket interface requires the client to send t
+            Group("nests") { 
+                ExampleNestComponent(birdID: $birdID)
+            }
         }
     }
 }
 
 TestWebService.main()
 ```
 
-*JSON Based APIs*
+## Explicit Options
 
-If the `Component` is registered at `/api/birds` the client can request a bird with the identifier `42` by sending a HTTP GET request to `/api/birds/42/`.
+Apodini web API exporters try to infer the best possible form of representing parameters.
+However, someweb API exporters may want to provide some additional customization, so that developers may forego the inference, and provide a strategy explicitely.
+The values of these options can be defined by the web API exporters, and each web API exporters can read for the options specific to itself.  
+For example: for HTTP JSON Based APIs, the exporter might want to allow the user to specify if a parameter is supposed to be retrieved from:
+- URI Path
+- HTTP Body
+- Query Params
+- etc.
 
-*gRPC*
+### Defining Options
 
-```protobuf
-service Birds {
-    rpc Example (BirdID) returns (...) {}
-}
+The exporter can define it's own options, by defining the type of the options, the key to identify the option and a helper to make it work seamlessly with the Parameter Property Wrapper:
 
-message BirdID {
-    int32 id = 1;
+```swift
+// Define a Type
+public enum HTTPParameterMode: PropertyOption {
+    case body
+    case path
+    case query
 }
-``` 
 
-*GraphQL*
-
-```graphql
-type Query {
-    example(birdID: Int): ...
+// Define a Key
+extension PropertyOptionKey where Property == ParameterOptionNameSpace, Option == HTTPParameterMode {
+    static let http = PropertyOptionKey<ParameterOptionNameSpace, HTTPParameterMode>()
 }
-```
 
-*WebSocket*
-
-Exposing the `Component` on a web socket interface requires the client to send the following message encoded as JSON:
-```json
-{
-    "type": "Example",
-    "parameters": {
-        "birdID": 42
+// Add helper to register it to a Parameter Property Wrapper
+extension AnyPropertyOption where Property == ParameterOptionNameSpace {
+    public static func http(_ mode: HTTPParameterMode) -> AnyPropertyOption<ParameterOptionNameSpace> {
+        return AnyPropertyOption(key: .http, value: mode)
     }
 }
 ```
 
-### Defining a Parameter  Defining an Endpoint outside an Endpoint
-
-The `@PathParameter` property can also be defined outside the component as part of a Group that contains the `Compoent`.
+With the new definition of `HTTPParameterMode` the developer may now define explicitely to the exporter how they would like their parameters to be exported:
 
 ```swift
-struct Bird: Identifiable {
-    var id: Int
-    var name: String
-    var age: Int
-}
-
-struct ExampleBirdComponent: Component {
-    @PathParameter var birdID: Bird.ID
+struct ExampleComponent: Component {
+    @Parameter(.http(.body)) 
+    var times: Int
+    
+    // Can define different handling for different exporters
+    @Parameter(.http(.path))
+    var id: UUID = true
```

Comment:
Is this still an outdated example, right? Cause `@Parameter(.http(.path))` was replaced by `@PathParameter`.
Additionally `@Parameters` should be placed on the Handler and `@PathParmeter` are the only ones placed at the Component level (?).